controllers/cluster_controller.go

```diff
@@ -50,6 +52,38 @@ const (
 	deleteRequeueAfter = 5 * time.Second
 )
 
+var (
+	clusterControlPlaneReady = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "capi_cluster_control_plane_ready",
+			Help: "Cluster control plane is ready if set to 1 and not if 0.",
+		},
+		[]string{"cluster", "namespace"},
+	)
+	clusterInfrastructureReady = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "capi_cluster_infrastructure_ready",
+			Help: "Cluster infrastructure is ready if set to 1 and not if 0.",
+		},
+		[]string{"cluster", "namespace"},
+	)
+	clusterKubeconfigReady = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "capi_cluster_kubeconfig_ready",
+			Help: "Cluster kubeconfig is ready if set to 1 and not if 0.",
+		},
+		[]string{"cluster", "namespace"},
+	)
+)
```

Comment:
Can we move these variables to a `metrics/` package under the controller folder, expose the variables and document them? This way, they'll show up in godoc as well